designer/client/conditions/SelectConditions.tsx

```diff
@@ -69,6 +77,106 @@ class SelectConditions extends React.Component<Props, State> {
       }, {});
   }
 
+  conditionsForPath(path: string) {
+    const { data } = this.context;
+    const fields: any = Object.values(this.fieldsForPath(path));
+    const { conditions = [] } = data;
+    var conditionsForPath: any[] = [];
+
+    const stringConditions = conditions.filter(
+      (condition) => typeof condition.value === "string"
+    );
+    const objectConditions = conditions.filter(
+      (condition) => typeof condition.value !== "string"
+    );
+
+    fields.forEach((field) => {
+      this.handleStringConditions(
+        stringConditions,
+        field.name,
+        conditionsForPath
+      );
+      this.handleConditions(objectConditions, field.name, conditionsForPath);
+    });
+
+    return conditionsForPath;
+  }
+
+  handleConditions(
+    objectConditions: ConditionData[],
+    fieldName: string,
+    conditionsForPath: any[]
+  ) {
+    objectConditions.forEach((condition) => {
+      condition.value.conditions.forEach((innerCondition) => {
+        this.checkAndAddCondition(
+          condition,
+          fieldName,
+          innerCondition.field.name,
+          conditionsForPath
+        );
+      });
+    });
+  }
+
+  handleStringConditions(
+    stringConditions: any[],
+    fieldName: string,
+    conditionsForPath: any[]
+  ) {
+    const operators = ["==", "!=", ">", "<"];
+    const conditionsWithAcceptedOperators = stringConditions.filter(
+      (condition) =>
+        operators.some((operator) => condition.value.includes(operator))
+    );
+    const conditionsWithFieldName = conditionsWithAcceptedOperators.map(
+      (condition) => ({
+        ...condition,
+        conditionFieldName: condition.value
+          .substring(
+            condition.value.indexOf(".") + 1,
+            condition.value.lastIndexOf(
+              operators.filter((operator) => condition.value.includes(operator))
+            )
+          )
+          .trim(),
+      })
+    );
+    conditionsWithFieldName.forEach((condition) =>
+      this.checkAndAddCondition(
+        condition,
+        fieldName,
+        condition.conditionFieldName,
+        conditionsForPath
+      )
+    );
+    var a = "";
+  }
+
+  checkAndAddCondition(
+    condition,
+    fieldName: string,
+    conditionFieldName: string,
+    conditions: any[]
+  ) {
+    if (this.checkDuplicateCondition(conditions, condition.name)) return;
+    if (fieldName == conditionFieldName) conditions.push(condition);
+  }
+
+  checkDuplicateCondition(conditions: any[], conditionName: string) {
+    for (const condition of conditions) {
+      if (condition.name == conditionName) return true;
+    }
+    return false;
+  }
```

Comment:
`===` is safer for js. I also don't think this needs it's own function. You can use 
```
const isDuplicateCondition = conditions.includes(condition => condition.name === conditionName)
```
includes explicitly returns true if the item was found, or false if not. (So you don't have to do if(..) return true)

Reply:
Method has no been removed